WordPress/Classes/Networking/TaxonomyServiceRemoteREST.m

```diff
@@ -2,76 +2,148 @@
 #import "WordPressComApi.h"
 #import "RemotePostCategory.h"
 #import "RemotePostTag.h"
+#import "RemoteTaxonomyPaging.h"
+
+static NSString * const TaxonomyServiceRemoteRESTCategoryTypeIdentifier = @"categories";
+static NSString * const TaxonomyServiceRemoteRESTTagTypeIdentifier = @"tags";
 
 @implementation TaxonomyServiceRemoteREST
 
 #pragma mark - categories
 
-- (void)getCategoriesWithSuccess:(void (^)(NSArray <RemotePostCategory *> *))success
-                         failure:(void (^)(NSError *))failure
-{
-    NSString *path = [NSString stringWithFormat:@"sites/%@/categories?context=edit", self.siteID];
-    NSString *requestUrl = [self pathForEndpoint:path
-                                     withVersion:ServiceRemoteRESTApiVersion_1_1];
-    
-    [self.api GET:requestUrl
-       parameters:nil
-          success:^(AFHTTPRequestOperation *operation, id responseObject) {
-              if (success) {
-                  success([self remoteCategoriesWithJSONArray:[responseObject arrayForKey:@"categories"]]);
-              }
-          } failure:^(AFHTTPRequestOperation *operation, NSError *error) {
-              if (failure) {
-                  failure(error);
-              }
-          }];
-}
-
 - (void)createCategory:(RemotePostCategory *)category
                success:(void (^)(RemotePostCategory *))success
                failure:(void (^)(NSError *))failure
 {
     NSParameterAssert(category.name.length > 0);
-    NSString *path = [NSString stringWithFormat:@"sites/%@/categories/new?context=edit", self.siteID];
-    NSString *requestUrl = [self pathForEndpoint:path
-                                     withVersion:ServiceRemoteRESTApiVersion_1_1];
     
     NSMutableDictionary *parameters = [NSMutableDictionary dictionary];
     parameters[@"name"] = category.name;
     if (category.parentID) {
         parameters[@"parent"] = category.parentID;
     }
+    
+    [self createTaxonomyWithType:TaxonomyServiceRemoteRESTCategoryTypeIdentifier
+                      parameters:parameters
+                         success:^(id responseObject) {
```

Comment:
Would it make sense to declare `responseObject` as an `NSDictionary` instead of `id` ?